tools/windows-node-installer/test/e2e/clustersetup.go

```diff
@@ -0,0 +1,195 @@
+	clientset "github.com/openshift/client-go/config/clientset/versioned"
+	"github.com/openshift/windows-machine-config-bootstrapper/tools/windows-node-installer/pkg/client"
+	corev1 "k8s.io/api/core/v1"
+	rbac "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	coreclient "k8s.io/client-go/kubernetes/typed/core/v1"
+	rbacclient "k8s.io/client-go/kubernetes/typed/rbac/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/auth"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	// configDir is a directory containing all yaml files for the resources
+	configDir = "clusterconfig"
+	// projectName represents namespace where the resources should be created
+	projectName = "windows-node-installer"
+	// resourceName represents name of the resources to be created
+	resourceName = "windows-node-installer"
+)
+
+// configFiles contains name of the resource files
+var configFiles = [...]string{"namespace.yaml", "role.yaml", "clusterole.yaml", "service_account.yaml", "cluster_role_binding.yaml", "role_binding.yaml"}
+
+// setRbacConfig creates role, clusterole, rolebinding and cluster role binding for the default service account in
+// cluster. It returns error if any of the resource creation fails.
+func setRbacConfig(cfg *rest.Config) error {
+	rbacConfig, err := rbacclient.NewForConfig(cfg)
+	if err != nil {
+		return err
+	}
+	core, _ := coreclient.NewForConfig(cfg)
+
+	for _, f := range configFiles {
+
+		bytes, err := ioutil.ReadFile(configDir + "/" + f)
+		if err != nil {
+			return fmt.Errorf("error reading the yaml file for the resource %v", err)
+		}
+		// UniversalDeserializer converts the yaml file format into Go objects recognized by k8s go-client
+		decode := authscheme.Codecs.UniversalDeserializer().Decode
+
+		obj, _, err := decode(bytes, nil, nil)
+
+		if err != nil {
+			return fmt.Errorf("error decoding YAML objects: %s", err)
+		}
+
+		switch o := obj.(type) {
+
+		case *corev1.Namespace:
+			// check if the namespace already exists
+			_, err := core.Namespaces().Get(projectName, metav1.GetOptions{})
+			if err != nil {
+				_, err := core.Namespaces().Create(o)
+				if err != nil {
+					return fmt.Errorf("error creating  namespace: %v", err)
+				}
+			}
+			log.Print("Namespace Created")
+
+		case *rbac.ClusterRole:
+			// check if the role already exists
+			_, err := rbacConfig.ClusterRoles().Get(resourceName, metav1.GetOptions{})
+			if err != nil {
+				_, err := rbacConfig.ClusterRoles().Create(o)
+				if err != nil {
+					return fmt.Errorf("error creating cluster role: %v", err)
+				}
+				log.Print("Cluster Role Created")
+			}
+
+		case *rbac.Role:
+			// check if the cluster role already exists
+			_, err := rbacConfig.Roles(projectName).Get(resourceName, metav1.GetOptions{})
+			if err != nil {
+				_, err := rbacConfig.Roles(projectName).Create(o)
+				if err != nil {
+					return fmt.Errorf("error creating  role: %v", err)
+				}
+				log.Print("Role Created")
+			}
+
+		case *corev1.ServiceAccount:
```

Comment:
If you are using the default SA, why are you creating one here?

Reply:
Yeah a difference in the function comment and the functionality.
I suggest changing the comment and keeping the creation of a new service account.

Reply:
Yes. I will update the comment and will keep a new service account.